lib/parser/provenance.cc

```diff
@@ -306,6 +306,34 @@ const SourceFile *AllSources::GetSourceFile(
       origin.u);
 }
 
+std::optional<SourcePosition> AllSources::GetSourcePosition(
+    Provenance prov) const {
+  const Origin &origin{MapToOrigin(prov)};
+  if (const auto *inc{std::get_if<Inclusion>(&origin.u)}) {
+    std::size_t offset{origin.covers.MemberOffset(prov)};
+    return inc->source.FindOffsetLineAndColumn(offset);
+  } else {
+    return std::nullopt;
+  }
+}
+
+std::optional<ProvenanceRange> AllSources::GetFirstFileProvenance() const {
+  for (const auto &origin : origin_) {
+    if (std::holds_alternative<Inclusion>(origin.u)) {
+      return origin.covers;
+    }
+  }
+  return std::nullopt;
+}
+
+ProvenanceRange AllSources::GetContiguousRangeAround(
```

Comment:
I don't think that this member function is ever used.

Reply:
My bad, this was incorrectly merged.